spec/index.bs

```diff
@@ -2748,82 +2800,81 @@ A {{GPUBindGroup}} object has the following internal slots:
+                                    - |resource| is a {{GPUTextureView}}.
                                     - |resource| is [$valid to use with$] |this|.
+                                    - Let |texture| be |resource|'s texture.
                                     - |layoutBinding|.{{GPUBindGroupLayoutEntry/viewDimension}} is
-                                        equal to |resource|'s {{GPUTextureViewDescriptor/dimension}}.
-                                    - |layoutBinding|.{{GPUBindGroupLayoutEntry/textureComponentType}}
-                                        is compatible with |resource|'s {{GPUTextureViewDescriptor/format}}.
-                                    - |resource|'s texture's {{GPUTextureDescriptor/usage}} includes
-                                            {{GPUTextureUsage/SAMPLED}}.
-
-                                - If |layoutBinding|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"sampled-texture"}}:
-                                    - |resource|'s texture's {{GPUTextureDescriptor/sampleCount}} is 1.
-
-                                - If |layoutBinding|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"multisampled-texture"}}:
-                                    - |resource|'s texture's {{GPUTextureDescriptor/sampleCount}} is greater than 1.
-
-                                - If |layoutBinding|.{{GPUBindGroupLayoutEntry/type}} is
-                                    {{GPUBindingType/"readonly-storage-texture"}} or
-                                    {{GPUBindingType/"writeonly-storage-texture"}}:
-                                    - |resource| is [$valid to use with$] |this|.
-                                    - |layoutBinding|.{{GPUBindGroupLayoutEntry/viewDimension}} is
-                                        equal to |resource|'s {{GPUTextureViewDescriptor/dimension}}.
-                                    - |resource|.{{GPUTextureView/[[descriptor]]}}.{{GPUTextureViewDescriptor/format}}
-                                        is equal to |layoutBinding|.{{GPUBindGroupLayoutEntry/storageTextureFormat}}.
-                                    - |resource|'s texture's {{GPUTextureDescriptor/usage}} includes
-                                        {{GPUTextureUsage/STORAGE}}.
-                                    - |resource|'s texture's {{GPUTextureDescriptor/sampleCount}} is 1.
-
-                                - If |layoutBinding|.{{GPUBindGroupLayoutEntry/type}} is
-                                    {{GPUBindingType/"uniform-buffer"}} or
-                                    {{GPUBindingType/"storage-buffer"}} or
-                                    {{GPUBindingType/"readonly-storage-buffer"}}:
+                                        equal to |resource|'s '{{GPUTextureViewDescriptor/dimension}}.
+                                    - If |bindingType| is {{GPUBindingType/"multisampled-texture"}}
```

Comment:
why aren't we merging all these conditions into a giant switch?

Reply:
LOL. I considered it but was worried y'all would think I'd gone overboard with the switch-ification. I gave it a try in the most recent commit to see how it works out, let me know what you think! (The indentation level is getting pretty deep.)

Reply:
we welcome our new switch overlords!